ipam/calico-ipam.go

```diff
@@ -222,24 +219,43 @@ func cmdDel(args *skel.CmdArgs) error {
 		return err
 	}
 
-	// Release the IP address by using the handle - which is workloadID.
+	// Release the IP address by using the handleID.
+	handleID := fmt.Sprintf("%s.%s", conf.Name, args.ContainerID)
```

Comment:
Are we worried about `conf.Name` containing invalid characters?

The CNI spec says this field is a "string" but doesn't give any more detail than that. Do we actually need to hash this or do some escaping to ensure we don't try to write a bad value?

Reply:
I think we do, if we don't really care how pretty it is hashing it seems like the better option to me, but if we want it to look sensible we should escape it, do we already have a function for this in libcalico somewhere?

Reply:
I went with b64 encoding the network name, once the tests pass locally I'll push up the code for the next review.